contracts/contracts/core/GasTankManager.sol

```diff
@@ -0,0 +1,84 @@
+// SPDX-License-Identifier: BSL 1.1
+pragma solidity 0.8.26;
+
+import {Errors} from "../utils/Errors.sol";
+
+/// @title GasTankManager
+/// @notice Coordinates on-demand ETH Transfers to the RPC Service for EOA custodial gas tanks.
+/// @dev This contract implicitly trusts the RPC_SERVICE address.
+contract GasTankManager {
+    address public immutable RPC_SERVICE;
+    uint256 public immutable MINIMUM_DEPOSIT;
+
+    event FundsRecovered(address indexed owner, uint256 indexed amount);
+    event GasTankFunded(address indexed smartAccount, address indexed caller, uint256 indexed amount);
+
+    error FailedToRecoverFunds(address owner, uint256 amount);
+    error NotValidCaller(address caller);
+    error InvalidAmount();
+    error FailedToFundGasTank(address rpcProvider, uint256 transferAmount);
+    error RPCServiceNotSet(address provider);
+    error NotRPCService(address caller);
+    error InsufficientFunds(uint256 currentBalance, uint256 requiredBalance);
+    error NotThisEOA(address msgSender, address thisAddress);
+    error MinimumDepositNotMet(uint256 amountToTransfer, uint256 minimumDeposit);
+
+    modifier onlyThisEOA() {
+        require(msg.sender == address(this), NotThisEOA(msg.sender, address(this)));
+        _;
+    }
+
+    modifier onlyRPCService() {
+        require(msg.sender == RPC_SERVICE, NotRPCService(msg.sender));
+        _;
+    }
+
+    constructor(address rpcService, uint256 _minDeposit) {
+        require(rpcService != address(0), RPCServiceNotSet(rpcService));
+        require(_minDeposit > 0, MinimumDepositNotMet(0, _minDeposit));
+        RPC_SERVICE = rpcService;
+        MINIMUM_DEPOSIT = _minDeposit;
+    }
+
+    receive() external payable { /* ETH transfers allowed. */ }
+
+    fallback() external payable {
+        revert Errors.InvalidFallback();
+    }
+
+    /// @notice Recovers funds inadvertently sent to this contract directly.
+    function recoverFunds() external onlyRPCService {
```

Comment:
Not sure what this function is for. This contract is running as the user's EOA as owner. So not sure why we need to recover funds?